control_plane/attachment_service/src/service.rs

```diff
@@ -753,13 +855,28 @@ impl Service {
                 };
 
                 this.startup_reconcile().await;
-
                 drop(startup_completion);
+            }
+        });
 
+        tokio::task::spawn({
+            let this = this.clone();
+            let startup_complete = startup_complete.clone();
+            async move {
+                startup_complete.wait().await;
                 this.background_reconcile().await;
             }
         });
 
+        tokio::task::spawn({
```

Comment:
We're spawning two tasks here: one in Service and one in HeartbeatHandler -- did you think about collapsing this into one task?

Reply:
We can collapse this to just the task in Service if you feel strongly about it, but I like the separation. The heartbeat state would have to move into the heartbeat related struct owned by the service. This would mean that the service holds some state outside of the `ServiceState` struct and it muddies the waters.

Reply:
Don't feel super strongly.